### PR TITLE
refactor: Use package.json for version management

### PR DESCRIPTION
**Pull request by DarhkVoyd**
## Description
- This PR updates the project configuration to use the package json file for version management. The change involves configuring reading the ```hatch version``` number from package.json using a regex pattern. 
- Additionally, it removes the logic that creates or updates the version.txt file, which is now redundant.
- Many files used ```pkgutil``` to extract the version, which lead to multiple similar errors after making the changes and deleting the version.txt files. Replaced it by loading version through package.json

Closes: #5489

### CODE REVIEW

**Comment by abidlabs** [comment]
Thanks for opening this PR @DarhkVoyd!

There's a few issues that we'll need to fix

1. There's a lot of code duplication across files -- we should consolidate into a single function (ideally located in `gradio_client/utils.py` since that can be imported from `gradio` as well)

2. The `package.json` won't be accessible directly using `open("package.json")` since its located inside the directory where `gradio` is installed. You'll need to do something like: `pkgutil.get_data(__name__, "package.json")`

3. If for some reason, the version cannot be fetched, we should *not* raise an error since this will prevent anyone from being able to use Gradio. Instead it should silently set the value of version to an empty string as in the original logic:

```py
# __version__ = (pkgutil.get_data(__name__, "version.txt") or b"").decode("ascii").strip()
```

**Comment by DarhkVoyd** [comment]
@abidlabs I have made changes and removed the code duplication by creating a function in utils as you asked.

**Comment by abidlabs** [comment]
Looks good thanks @DarhkVoyd! Let me fix the conflict and just confirm everything works locally

File: gradio/__init__.py [hunk 1]
```diff
@@ -104,7 +106,4 @@
 )
 from gradio.themes import Base as Theme
 
-current_pkg_version = (
-    (pkgutil.get_data(__name__, "version.txt") or b"").decode("ascii").strip()
-)
-__version__ = current_pkg_version
```
Comment (abidlabs) [inline gradio/__init__.py:110]: We should keep `__version__` as doing `gradio.__version__` is the canonical way to get the version of a package

File: gradio/__init__.py [hunk 2]
```diff
@@ -104,7 +106,4 @@
...
+current_pkg_version = get_package_version()
```
Comment (abidlabs) [inline gradio/__init__.py:109]: This gets the gradio_client version even more the main gradio package

**Comment by abidlabs** [comment]
Noticed a couple of small things -- I'll push the fixes so that we can get this merged in.

**Comment by abidlabs** [comment]
There were some other places that we used `version.txt` -- I've updated the other references. This should be good to go, I'll merge later today & do some more testing

**Comment by abidlabs** [comment]
Merged, thanks for the contribution @DarhkVoyd!

**Pull request merged**